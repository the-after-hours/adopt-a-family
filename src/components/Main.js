import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Landing from './Landing';
import Dashboard from './Dashboard';
import WishlistPublic from './WishlistPublic';
import WishlistPrivate from './WishlistPrivate';
import AboutUs from './AboutUs';
import Pairing from './Pairing';
import Messaging from './Messaging';
import SecureOrg from './SecureOrg';
import Nav from './Nav';
import Footer from './Footer';

const Main = (props) => {
  return (
    <div>
      <Nav />
      <Switch>
        <Route exact path="/" component={Landing} />
        <Route path="/dashboard" component={Dashboard} />
        <Route path="/wishlist" component={WishlistPublic} />
        <Route path="/aboutus" component={AboutUs} />
        <Route path="/secure/wishlist" component={WishlistPrivate} /> {/* Secure */}
        <Route path="/secure/pairing" component={Pairing} /> {/* Secure */}
        <Route path="/secure/messaging" component={Messaging} /> {/* Secure */}
        <Route path="/secure/org" component={SecureOrg} /> {/* Secure */}
      </Switch>

      <Footer />
    </div>
  );
};

export default Main;
