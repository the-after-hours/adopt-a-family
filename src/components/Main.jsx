import React from 'react';
import { Route, Switch } from 'react-router-dom';

import AboutUs from './AboutUs';
import Dashboard from './Dashboard';
import Footer from './Footer';
import Landing from './Landing';
import Messaging from './Messaging';
import Header from './Header';
import Pairing from './Pairing';
import Registration from './Registration';
import SecureOrg from './SecureOrg';
import WishlistPrivate from './WishlistPrivate';
import WishlistPublic from './WishlistPublic';

const Main = (props) => (
  <div>
    <Header />
    <Switch>
      <Route exact path="/" component={Landing} />
      <Route path="/aboutus" component={AboutUs} />
      <Route path="/dashboard" component={Dashboard} />
      <Route path="/secure/messaging" component={Messaging} /> {/* Secure */}
      <Route path="/secure/org" component={SecureOrg} /> {/* Secure */}
      <Route path="/secure/pairing" component={Pairing} /> {/* Secure */}
      <Route path="/secure/wishlist" component={WishlistPrivate} />{' '}
      {/* Secure */}
      <Route path="/registration" component={Registration} />
      <Route path="/wishlist" component={WishlistPublic} />
    </Switch>

    <Footer />
  </div>
);

export default Main;
