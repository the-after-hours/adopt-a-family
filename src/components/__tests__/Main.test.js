import React from 'react';
import Main from '../main.js';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('Main', () => {
  test('Main snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Main />
      </MemoryRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});