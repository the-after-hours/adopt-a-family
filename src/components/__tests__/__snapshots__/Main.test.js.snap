// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Main snapshot 1`] = `
<div>
  <div>
    <h1>
      Landing page
    </h1>
    <ul>
      <li>
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          Click me for dashboard
        </a>
      </li>
      <li>
        <a
          href="/secure/pairing"
          onClick={[Function]}
        >
          Click me for secure/pairing
        </a>
      </li>
      <li>
        <a
          href="/secure/messaging"
          onClick={[Function]}
        >
          Click me for messaging
        </a>
      </li>
      <li>
        <a
          href="/wishlist"
          onClick={[Function]}
        >
          Click me for wishlistPublic
        </a>
      </li>
      <li>
        <a
          href="/secure/wishlist"
          onClick={[Function]}
        >
          Click me for secure/wishlistPrivate
        </a>
      </li>
      <li>
        <a
          href="/secure/org"
          onClick={[Function]}
        >
          Click me for secure/org
        </a>
      </li>
      <li>
        <a
          href="/aboutus"
          onClick={[Function]}
        >
          Click me for about us
        </a>
      </li>
    </ul>
  </div>
  <footer>
    <a
      href="/"
      onClick={[Function]}
    >
      <button>
        Home
      </button>
    </a>
  </footer>
</div>
`;
